src/hotspot/cpu/aarch64/sharedRuntime_aarch64.cpp

```diff
@@ -2021,7 +2027,11 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
     __ mov(c_rarg2, rthread);
 
     // Not a leaf but we have last_Java_frame setup as we want
+    // Force freeze slow path in case we try to preempt. We will pin the
+    // vthread to the carrier (see FreezeBase::recurse_freeze_native_frame()).
+    __ push_cont_fastpath();
```

Comment:
Right. We want to take the slow path to find the compiled native wrapper frame and fail to freeze. Otherwise the fast path won't find it since we don't walk the stack.

Reply:
It would be nice if Coleen's question and your answer could be turned into a source comment. It really describes what's going more clearly than the current comment.

Reply:
I updated the comment.